src/messaging/ExchangeMgr.h

```diff
@@ -148,6 +159,17 @@ class DLL_EXPORT ExchangeManager : public SecureSessionMgrDelegate
      */
     CHIP_ERROR UnregisterUnsolicitedMessageHandler(uint32_t protocolId, uint8_t msgType);
 
+    /**
+     * A strongly-message-typed version of UnregisterUnsolicitedMessageHandler.
+     */
+    template <typename MessageType, typename = std::enable_if_t<std::is_enum<MessageType>::value>>
+    CHIP_ERROR UnregisterUnsolicitedMessageHandler1(MessageType msgType)
```

Comment:
The 1 suffix is a bit ugly.

How about something like:

`UnregisteerUnsolicitedMessagesHandlerForProtodcol` and `UnregisterUnsolicitedMessageHandlerForType`

Overloads for non-equivalent types are very hard to follow.